### PR TITLE
Dynamically set y-axis label width for credits chart

### LINKED ISSUE #11586: Credits chart axis should calculate dynamic width
The credit consumption chart currently has a fixed width causing the tick labels to be cut off depending on their length. The axis width should be calculated dynamically.

screenshot-20220330-172907

### PR DESCRIPTION
**Pull request by krishnaglick**
Fixes #11586

This is honestly a little hackier than I wanted, but we don't have a way to dynamically set the [width](https://recharts.org/en-US/api/YAxis#width)

### CODE REVIEW

File: airbyte-webapp/src/components/BarChart/BarChart.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-import { CartesianGrid, BarChart as BasicBarChart, ResponsiveContainer, XAxis, YAxis, Bar, Label } from "recharts";
+import { Bar, BarChart as BasicBarChart, CartesianGrid, Label, ResponsiveContainer, XAxis, YAxis } from "recharts";
 import { barChartColors, theme } from "theme";
 
 type BarChartProps = {
-  data?: {
+  data: {
     name: string;
-    value: number | number[];
+    value: number;
```
Comment (edmundito) [inline airbyte-webapp/src/components/BarChart/BarChart.tsx:8]: Is it safe to remove the number array? How come the type was introduced?

File: airbyte-webapp/src/components/BarChart/BarChart.tsx [hunk 2]
```diff
@@ -16,6 +16,11 @@ const BarChart: React.FC<BarChartProps> = ({ data, legendLabels, xLabel, yLabel
   const chartLinesColor = theme.greyColor20;
   const chartTicksColor = theme.lightTextColor;
 
+  const width = Math.min(
```
Comment (edmundito) [inline airbyte-webapp/src/components/BarChart/BarChart.tsx:19]: There's some sorting happening. I think it deserves a memo

File: airbyte-webapp/src/components/BarChart/BarChart.tsx [hunk 3]
```diff
@@ -16,6 +16,11 @@ const BarChart: React.FC<BarChartProps> = ({ data, legendLabels, xLabel, yLabel
...
+    Math.max([...data].sort((a, b) => b.value - a.value)[0].value.toFixed(0).length * 10, 80),
```
Comment (edmundito) [inline airbyte-webapp/src/components/BarChart/BarChart.tsx:20]: Assumes that there's always at least 1 item in the array?

File: airbyte-webapp/src/components/BarChart/BarChart.tsx (hunk 1 above)
Comment (krishnaglick) [reply airbyte-webapp/src/components/BarChart/BarChart.tsx:8]: This component is only used in one place, and the types now match the incoming data.

File: airbyte-webapp/src/components/BarChart/BarChart.tsx (hunk 3 above)
Comment (krishnaglick) [reply airbyte-webapp/src/components/BarChart/BarChart.tsx:20]: The default data is an array of dates + values at 0.

**Review by edmundito — APPROVED** [review]

**Pull request merged**